src/sentry/services/hybrid_cloud/organization_mapping/impl.py

```diff
@@ -25,6 +30,58 @@ def get_many(self, *, organization_ids: List[int]) -> List[RpcOrganizationMappin
         org_mappings = OrganizationMapping.objects.filter(organization_id__in=organization_ids)
         return [serialize_organization_mapping(om) for om in org_mappings]
 
+    def _check_organization_mapping_integrity(
+        self, org_id: int, update: RpcOrganizationMappingUpdate
+    ) -> bool:
+        if not update.slug:
+            capture_exception(
+                OrganizationMappingConsistencyException("Organization mapping must have a slug")
+            )
+            return False
+
+        if not update.region_name:
+            capture_exception(
+                OrganizationMappingConsistencyException("Organization mapping must have a region")
+            )
+            return False
+
+        org_slug_qs = OrganizationSlugReservation.objects.filter(
+            organization_id=org_id,
+        )
+        org_slugs = [org_slug for org_slug in org_slug_qs]
+
+        if len(org_slugs) == 0:
+            # If there's no matching organization slug reservation, alert but don't prevent writing a new org mapping
+            capture_exception(
+                OrganizationMappingConsistencyException(
+                    f"Expected an organization slug reservation for organization {org_id}, none was found"
+                )
+            )
+            return False
```

Comment:
Does returning false here prevent the mapping update? Should the comment be updated?

Reply:
Ahh yep, I just flipped this and forgot to update the comment 😅 Thanks!